src/NHSD.BuyingCatalogue.Solutions.Application/Persistence/AdditionalServiceReader.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using NHSD.BuyingCatalogue.Solutions.Application.Domain;
+using NHSD.BuyingCatalogue.Solutions.Contracts.Persistence;
+
+namespace NHSD.BuyingCatalogue.Solutions.Application.Persistence
+{
+    internal sealed class AdditionalServiceReader
+    {
+        private readonly IAdditionalServiceRepository _additionalServiceRepository;
+
+        public AdditionalServiceReader(IAdditionalServiceRepository additionalServiceRepository)
+        {
+            _additionalServiceRepository = additionalServiceRepository;
```

Comment:
Throw if null? Or did we stop doing that in BAPI?

Reply:
I cant seem to find any examples where we have thrown in the reader, but I can add it in if you want

Reply:
if we don't do it in BAPI, it's all good